ui/test/services/project_service_test.dart

```diff
@@ -36,20 +36,32 @@ void main() {
     });
 
     group('Projects query', () {
-      test('queries list of projects', () async {
-        when(client.getProjects())
-            .thenAnswer((_) => Future.value([Project(id: projectId)]));
+      test('query to list all projects when no filter applied', () async {
+        when(client.findProjectsBySearchFragment(''))
```

Comment:
Suggestion is to pass a fragment (instead of the empty string), so you the test shows this is what is passed between this call and the REST client.